src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs

```diff
@@ -175,13 +175,14 @@ BEGIN TRANSACTION
                         WITH (Uri varchar(128) '$.Uri')
                         EXCEPT SELECT Uri FROM dbo.SearchParam;
 
-                        IF ((SELECT COUNT(*) FROM dbo.SearchParamStatusRegistry) = 0)
+                        IF ((SELECT COUNT(*) FROM dbo.SearchParam WHERE Status IS NOT NULL) = 0)
                         BEGIN
                             DECLARE @lastUpdated datetimeoffset(7) = SYSDATETIMEOFFSET()
         
-                            INSERT INTO dbo.SearchParamStatusRegistry (Uri, Status, LastUpdated, IsPartiallySupported)
-                            SELECT sps.Uri, sps.Status, @lastUpdated, sps.IsPartiallySupported
-                            FROM @searchParamStatuses AS sps
+                            UPDATE dbo.SearchParam
+                            SET Status = sps.Status, LastUpdated = @lastUpdated, IsPartiallySupported = sps.IsPartiallySupported
+                            FROM dbo.SearchParam INNER JOIN @searchParamStatuses as sps
+                            ON dbo.SearchParam.Uri = sps.Uri
```

Comment:
@johnstairs is there a nicer way to do this?

Reply:
I'm not up to date on the design, but you want to update these if none of the statuses have been set?

Reply:
You're right. I don't think these values should ever be `null`. I made changes in 4bf9d07 to ensure this.